sympy/solvers/solveset.py

```diff
@@ -572,9 +576,10 @@ def _solveset(f, symbol, domain, _check=False):
         # wrong solutions we are using this technique only if both f and g are
         # finite for a finite input.
         result = Union(*[solver(m, symbol) for m in f.args])
-    elif _is_function_class_equation(TrigonometricFunction, f, symbol) or \
-            _is_function_class_equation(HyperbolicFunction, f, symbol):
+    elif _is_function_class_equation(TrigonometricFunction, f, symbol):
         result = _solve_real_trig(f, symbol)
+    elif _is_function_class_equation(HyperbolicFunction, f, symbol):
+        result = _solve_hyperbolic(f, symbol)
```

Comment:
I intend to separate out the solver for hyperbolic expressions from that of trigonometric solver since solving via `_solve_real_trig` returns a `ConditionSet` for even trivial hyperbolic expressions.
So, I think merging #10709 will be independent of this because it doesn't address this issue.

Moreover, wouldn't it be a better code encapsulation if we check for the `domain` intersection in `_solveset` as compared to individual solvers ?

What are your thoughts on this ?